plugins/modules/purefa_network.py

```diff
@@ -227,35 +266,106 @@ def update_fc_interface(module, array, interface, api_version):
     module.exit_json(changed=changed)
 
 
+def _create_subordinates(module, array, init=False):
+    subordinates = []
+    all_children = True
+    if module.params["subordinates"]:
+        for inter in module.params["subordinates"]:
+            if array.get_network_interfaces(names=[inter]).status_code != 200:
+                all_children = False
+            if not all_children:
+                module.fail_json(
+                    msg="Subordinate {0} does not exist. Ensure you have specified the controller.".format(
+                        inter
+                    )
+                )
+            if init:
+                subordinates.append(FixedReferenceNoId(name=inter))
+            else:
+                subordinates.append(inter)
+    return subordinates
+
+
+def _create_subinterfaces(module, array, init=False):
+    subinterfaces = []
+    all_children = True
+    purity_vm = bool(len(array.get_controllers().items) == 1)
+    if module.params["subinterfaces"]:
+        for inter in module.params["subinterfaces"]:
+            # As we may be on a single controller device, only check for the ct0 version of the interface
+            if array.get_network_interfaces(names=["ct0." + inter]).status_code != 200:
+                all_children = False
+            if not all_children:
+                module.fail_json(
+                    msg="Child subinterface {0} does not exist".format(inter)
+                )
+            if init:
+                subinterfaces.append(FixedReferenceNoId(name="ct0." + inter))
+            else:
+                subinterfaces.append("ct0." + inter)
+            if not purity_vm:
+                if init:
+                    subinterfaces.append(FixedReferenceNoId(name="ct1." + inter))
+                else:
+                    subinterfaces.append("ct1." + inter)
+    return subinterfaces
+
+
 def update_interface(module, array, interface):
     """Modify Interface settings"""
     changed = False
     current_state = {
+        "enabled": interface["enabled"],
         "mtu": interface["mtu"],
         "gateway": interface["gateway"],
         "address": interface["address"],
         "netmask": interface["netmask"],
         "services": sorted(interface["services"]),
+        "slaves": sorted(interface["slaves"]),
     }
+    array6 = get_array(module)
+    subinterfaces = current_state["slaves"]
+    if module.params["subinterfaces"]:
+        if sorted(module.params["subinterfaces"]) != current_state["slaves"]:
+            subinterfaces = _create_subinterfaces(module, array6)
+        else:
+            subinterfaces = current_state["slaves"]
```

Comment:
There are a couple of issues here. Assuming you are following the description of the `subinterfaces` parameter and only providing the "eth" number, the if statement will always equate to true. This is because `module.params["subinterfaces"]` returns just the eth numbers and `current_state["slaves"]` returns the full interface name including the controller. 

The second issue is that the `current_state["slaves"]` is sorted. but the response from `_create_subinterfaces` is not sorted. So later on when evaluating if the `current_state` and the `new_state` are the same, it always fails if you are using more than 1 interface on each controller. This could be fixed here when setting `subinterfaces` or in the return from the `_create_subinterfaces` function.

Reply:
True - Fixed